### DISCUSSION TITLE
Move `config/cache.c` to core library

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#3968] [Development]
The config code lives in the `config/` directory.
It is fairly self-contained; it only relies on helper functions from `libmutt`.

Except for two things: Config Cache, Address Type.

The Config Cache uses config functions, but it also relies on the `NeoMutt` object.
This means it should live in the `core/` directory.

This is a simple job to adjust the Makefile, but there's slightly more to do, to connect everything up.

If you like puzzles, I'll leave it at that.
(or I can provide a list)

**Comment by roccoblues** [comment]
Started here: https://github.com/neomutt/neomutt/pull/3989

**Comment by flatcap** [comment]
Completed in https://github.com/neomutt/neomutt/commit/21b377c4ea5557969d33090a83133d24036ad311
